From ba6f209b8b7aa6cc70f76942f402dae692be6b08 Mon Sep 17 00:00:00 2001
From: Zahari Doychev <zahari.doychev@intel.com>
Date: Tue, 22 Sep 2020 11:45:43 +0200
Subject: [PATCH] net: sched: tc-police api to get peak rate and mtu

Add functions to get the peak rate and mtu which makes possible to offload
dual rate policers. The upstream patches allow only offload of single rate
policers.

Signed-off-by: Zahari Doychev <zahari.doychev@intel.com>
---
 include/net/tc_act/tc_police.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/net/tc_act/tc_police.h b/include/net/tc_act/tc_police.h
index 41b0b823faa3..a911da72a586 100644
--- a/include/net/tc_act/tc_police.h
+++ b/include/net/tc_act/tc_police.h
@@ -54,4 +54,14 @@ static inline s64 tcf_police_tcfp_burst(const struct tc_action *act)
 	return to_police(act)->tcfp_burst;
 }
 
+static inline u64 tcf_police_peak_bytes_ps(const struct tc_action *act)
+{
+	return to_police(act)->peak.rate_bytes_ps;
+}
+
+static inline u32 tcf_police_tcfp_mtu(const struct tc_action *act)
+{
+	return to_police(act)->tcfp_mtu;
+}
+
 #endif /* __NET_TC_POLICE_H */
